### PR TITLE
Couple typos and `is_m4` function

### PR DESCRIPTION
**Pull request by christiangnrd**

### CODE REVIEW

File: lib/mtl/device.jl [hunk 1]
```diff
@@ -122,3 +122,4 @@ is_m1(dev::MTLDevice) = supports_family(dev, MTLGPUFamilyApple7) &&
 is_m2(dev::MTLDevice) = supports_family(dev, MTLGPUFamilyApple8) &&
                         !supports_family(dev, MTLGPUFamilyApple9)
 is_m3(dev::MTLDevice) = supports_family(dev, MTLGPUFamilyApple9)
+is_m4(dev::MTLDevice) = supports_family(dev, MTLGPUFamilyApple9)
```
Comment (maleadt) [inline lib/mtl/device.jl:125]: Wouldn't it be surprising for `is_m3` and `is_m4` to alias? Maybe we should do a match on `dev.name` here to make sure we can discern between both.

Comment (maleadt) [reply lib/mtl/device.jl:125]: Is it really the same GPU? Or just the same feature set? I wouldn't be surprised if the back-ends are at least somewhat different.

Comment (christiangnrd) [reply lib/mtl/device.jl]: Good point. I implemented the `dev.name` check

**Comment by christiangnrd** [comment]
@maleadt Do you know why the github actions benchmark test still triggers?

**Comment by maleadt** [comment]
No. Did you verify it worked before? Maybe I messed something up in https://github.com/JuliaGPU/Metal.jl/commit/6ecb9095a63a2bcfe7d0c10b5c6c72d408d1277f

**Comment by christiangnrd** [comment]
Not sure what the Permission Denied errors from mac mini 5.0 are about.

**Comment by maleadt** [comment]
> Not sure what the Permission Denied errors from mac mini 5.0 are about.

The compilecache files on there have really weird permissions:

```
julia@macmini-aarch64-5 ~ % ls -la /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-5.0/depots/5cd495a2-4a16-4674-ae02-c839447744bb/compiled/v1.11/Metal/ACDsk_tBsmV.ji
--w-------  1 julia  staff  21152 Dec 16 13:28 /Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-5.0/depots/5cd495a2-4a16-4674-ae02-c839447744bb/compiled/v1.11/Metal/ACDsk_tBsmV.ji
```

It's the only file with such permissions in there:

```
julia@macmini-aarch64-5 ~ % find /Users/julia/.julia/scratchspaces/*/agent-cache/*/depots/*/compiled -perm 200
/Users/julia/.julia/scratchspaces/a66863c6-20e8-4ff4-8a62-49f30b1f605e/agent-cache/default-macmini-aarch64-5.0/depots/5cd495a2-4a16-4674-ae02-c839447744bb/compiled/v1.11/Metal/ACDsk_tBsmV.ji
```

**Pull request merged**